pyEpiabm/pyEpiabm/tests/test_unit/test_core/test_person.py

```diff
@@ -110,6 +110,17 @@ def test_vaccinate(self):
         self.assertTrue(self.person.is_vaccinated)
         self.assertEqual(self.person.date_vaccinated, 5)
 
+    def test_set_id(self):
+        init_id = self.microcell.id + "." + "." + \
+                  str(len(self.microcell.persons) - 1)
+        self.assertEqual(self.person.id, init_id)
+        self.person.set_id("2.3.4.5")
+        self.assertEqual(self.person.id, "2.3.4.5")
+        self.assertRaises(TypeError, self.person.set_id, 2.0)
+        self.assertRaises(ValueError, self.person.set_id, "0.1")
+        self.assertRaises(ValueError, self.person.set_id, "1234")
+        self.assertRaises(ValueError, self.person.set_id, "0.0.0.0.5")
```

Comment:
These are great tests, but you should be consistent - in other tests you also check for errors with IDs that are too short (e.g. "0.0.0"), but you don't check for a series of numbers that are not period-separated. It would be good to have al these tests in all cases

Reply:
Sure, will do